networkx/drawing/tests/test_latex.py

```diff
@@ -51,15 +51,12 @@ def test_tikz_attributes():
     \end{tikzpicture}
 \end{figure}"""
 
+    # First, check for consistency line-by-line - if this fails, the mismatched
+    # line will be shown explicitly in the failure summary
+    for expected, actual in zip(expected_tex.split("\n"), output_tex.split("\n")):
+        assert expected == actual
```

Comment:
```suggestion
        print(f"-{expected}|\n+{actual}|")
```
This let's us see all the differences as part of the failure message...   But it is adding print statements to tests. :}

Reply:
The `assert` will also show you the differences as well: here's the output from the test after I changed a character in the "expected" output:

```
        # First, check for consistency line-by-line - if this fails, the mismatched
        # line will be shown explicitly in the failure summary
        for expected, actual in zip(expected_tex.split("\n"), output_tex.split("\n")):
>           assert expected == actual
E           AssertionError: assert '        \\dr...5] {2nd} (2);' == '        \\dr...5] {2nd} (2);'
E             
E             -         \draw[>->,bend right,line width=3,green!90] (1) to node[pos=0.5] {2nd} (2);
E             +         \draw[>->,bend rightt,line width=3,green!90] (1) to node[pos=0.5] {2nd} (2);
E             ?                            +

networkx/drawing/tests/test_latex.py:57: AssertionError
```

Is this good enough? Or are print statements still preferred?

Reply:
I tried a number of different error cases. 
It's pretty clear that it is better to stop at the first difference! :p

As just one example, the difference between
```E       AssertionError: assert '\\documentcl...end{document}' == '\\documentcl...end{document}'
E         Skipping 950 identical leading characters in diff, use -v to show
E           7);
E         +         \draw (5) to (2);
E                 \end{scope}
E               \end{tikzpicture}
E               \caption{My tikz number 1 of 2}\label{tikz_1_2}
E             \end{subfigure}...
E         
E         ...Full output truncated (37 lines hidden), use '-vv' to show
```
and what follows (in the "details") is that printing everything leads to information overload. A shift in a line, makes all the rest of the lines appear as errors. It's much better to focus on the first line that is different.
<details>

```
-      \end{scope}|
+        \draw (5) to (2);|
-    \end{tikzpicture}|
+      \end{scope}|
-    \caption{My tikz number 1 of 2}\label{tikz_1_2}|
+    \end{tikzpicture}|
-  \end{subfigure}|
+    \caption{My tikz number 1 of 2}\label{tikz_1_2}|
-  \begin{subfigure}{0.5\textwidth}|
+  \end{subfigure}|
-  \begin{tikzpicture}[scale=2]|
+  \begin{subfigure}{0.5\textwidth}|
-      \draw[gray!90]|
+  \begin{tikzpicture}[scale=2]|
-        (0.749, 0.702) node[green!90] (0){0}|
+      \draw[gray!90]|
-        (1.0, -0.014) node[green!90] (1){1}|
+        (0.749, 0.702) node[green!90] (0){0}|
-        (-0.777, -0.705) node (2){2}|
+        (1.0, -0.014) node[green!90] (1){1}|
-        (-0.984, 0.042) node (3){3}|
+        (-0.777, -0.705) node (2){2}|
-        (-0.028, 0.375) node[purple!90] (4){4}|
+        (-0.984, 0.042) node (3){3}|
-        (-0.412, 0.888) node (5){5}|
+        (-0.028, 0.375) node[purple!90] (4){4}|
-        (0.448, -0.856) node (6){6}|
+        (-0.412, 0.888) node (5){5}|
-        (0.003, -0.431) node[purple!90] (7){7};|
+        (0.448, -0.856) node (6){6}|
-      \begin{scope}[->,gray!90]|
+        (0.003, -0.431) node[purple!90] (7){7};|
-        \draw (0) to (4);|
+      \begin{scope}[->,gray!90]|
-        \draw (0) to (5);|
+        \draw (0) to (4);|
-        \draw (0) to (6);|
+        \draw (0) to (5);|
-        \draw (0) to (7);|
+        \draw (0) to (6);|
-        \draw (1) to (4);|
+        \draw (0) to (7);|
-        \draw (1) to (5);|
+        \draw (1) to (4);|
-        \draw (1) to (6);|
+        \draw (1) to (5);|
-        \draw (1) to (7);|
+        \draw (1) to (6);|
-        \draw (2) to (4);|
+        \draw (1) to (7);|
-        \draw (2) to (5);|
+        \draw (2) to (4);|
-        \draw (2) to (6);|
+        \draw (2) to (5);|
-        \draw (2) to (7);|
+        \draw (2) to (6);|
-        \draw (3) to (4);|
+        \draw (2) to (7);|
-        \draw (3) to (5);|
+        \draw (3) to (4);|
-        \draw (3) to (6);|
+        \draw (3) to (5);|
-        \draw (3) to (7);|
+        \draw (3) to (6);|
-      \end{scope}|
+        \draw (3) to (7);|
-    \end{tikzpicture}|
+        \draw (5) to (2);|
-    \caption{My tikz number 2 of 2}\label{tikz_2_2}|
+      \end{scope}|
-  \end{subfigure}|
+    \end{tikzpicture}|
-  \caption{A graph generated with python and latex.}|
+    \caption{My tikz number 2 of 2}\label{tikz_2_2}|
-\end{figure}|
+  \end{subfigure}|
-\end{document}|
+  \caption{A graph generated with python and latex.}|
```

</details>

So the PR is better than the commented out code!
Thanks for helping me see this. :)